torch/quantization/observer.py

```diff
@@ -284,6 +284,7 @@ def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
     def forward(self, x_orig):
         r"""Records the running minimum and maximum of ``x``."""
         x = x_orig.detach()  # avoid keeping autograd tape
+        x = x.to(self.min_val.dtype)
```

Comment:
Why is this needed?

Reply:
on lines 276/277, the min and max val buffers are initialized with the default floating point data type.  Since all updates are done in place, comparisons will crash unless the data types of both tensors being compared are equal.